### PR TITLE
Redundant cooling fan synced to FAN0

### PR DESCRIPTION
**Pull request by ellensp**
### Description

Add dual part cooling fans. 
Where the user has two part cooling fans on two IO ports and wants them to act as one (without soldering them together) 
I have seen this requested a number of time, and have talked a few through editing the code to add it.
So I'm adding it as a feature 

### Requirements

A number of controllable fans

### Benefits

Can control two part cooling fans with one M106/M107 gcode  (without needing to change gcode at all)

### Configurations

Added the following to Configuration_adv.h 

```
/**
 * Enable a second fan for part cooling
 */
//#define DUAL_PART_COOLING_FANS
#if ENABLED(DUAL_PART_COOLING_FANS)
  #define DUAL_PART_COOLING_FAN 2 // Define the number of the fan to use, eg FAN2 set 2, FAN1 set 1
#endif
```

### Related Issues
Latest request from 死Xtyling on discord.

### CODE REVIEW

**Review by HxYxI — APPROVED** [review]

**Comment by shitcreek** [comment]
could you reword it to be more precise, something like enable omni fan control?

**Comment by DrumClock** [comment]
Hi @ellensp , dobrá práce!
Měl bych prosbu, zda je možné dodělat také dva "automatické" ventilátory pro systém IDEX.
Kdy vozík X1 a X2 má svůj vlastní ventilátor pro chlazení tisku. A když zrovna parkuje, tak by se ventilátor vypnul.
Bohužel můj Slicer neumí rozlišit M106 Px na základě použitého nástroje Tx. V g-codu je jen M106 Sxxx a je jedno
který nástroj Tx je používán.
A v současném FW jsem možnost automatického přepínání ventilátoru (P0,P1) v závislosti vybraného nástroje (T0/T1) nenašel.

Např.
```
#define X1_PART_COOLING_FANS     FAN_PIN                           // Part-Cooling Fan X1
#define X2_PART_COOLING_FANS     FAN1_PIN                         // Part-Cooling Fan X2
```

Děkuji za informace.

**Comment by ellensp** [comment]
@shitcreek I was thinking dual, like dual z (both get same step/direction) or  X_DUAL_STEPPER_DRIVERS etc, so dual part  cooling fans seems to me the obvious name for two fans acting together as one

**Comment by thinkyhead** [comment]
Adjusted the code so that once a fan has been tied to FAN0 it always synchronizes and can never be adjusted separately.

**Comment by Russ48** [comment]
There was more to this. I finally got it to work with SKR Mini 3 V2.0 with two fan ports on an Ender 3 MAX. Controller fan hard wired to 24 volt output as configured from stock. Had to disable Controller in ADV.h.  the enable redundant cooling fan. Had to change  the assignment (but renamed) of pin PC7 from the Controller to The Redundant Cooling Fan. Now with M106/107 the part cooling fans come on and go off with M107...
